### PR TITLE
STOR-1573: Rebase `node-driver-registrar` to v2.10.0 for OCP 4.16

### PR DESCRIPTION
**Pull request by mpatlasov**
Diff to upstream v2.10.0:
https://github.com/kubernetes-csi/node-driver-registrar/compare/v2.10.0...mpatlasov:rebase-v2.10.0

Changes between 2.9.0 (OCP 4.15) and 2.10.0:

 * Update kubernetes dependencies to v1.29.0 (#366, @sunnylovestiramisu)

Full changelogs:

https://github.com/kubernetes-csi/node-driver-registrar/releases/tag/v2.10.0

@openshift/storage

### CODE REVIEW

**Comment by mpatlasov** [comment]
`ci/prow/security` job failure is false-positive: it complains about "Unsanitized input from a CLI argument...", but only admin can modify this input. We can safely ignore this `ci/prow/security` job failure.

File: go.mod [hunk 1]
```diff
@@ -40,16 +39,17 @@ require (
 	github.com/prometheus/common v0.44.0 // indirect
 	github.com/prometheus/procfs v0.10.1 // indirect
 	github.com/spf13/pflag v1.0.5 // indirect
-	go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc v0.46.1 // indirect
-	go.opentelemetry.io/otel v1.21.0 // indirect
-	go.opentelemetry.io/otel/metric v1.21.0 // indirect
-	go.opentelemetry.io/otel/trace v1.21.0 // indirect
-	golang.org/x/net v0.19.0 // indirect
+	go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc v0.46.0 // indirect
```
Comment (jsafrane) [inline go.mod:42]: This goes back in time, but it should not break anything (esp. it does not re-introduce CVE-2023-45142 nor CVE-2023-47108)

**Comment by Phaow** [comment]
/label qe-approved

CI looks good. The failed security check is not caused by this rebase.

**Pull request merged**